common/src/api/external/mod.rs

```diff
@@ -1966,26 +2078,41 @@ pub struct L4PortRange {
 }
 
 impl FromStr for L4PortRange {
-    type Err = String;
+    type Err = Error;
     fn from_str(range: &str) -> Result<Self, Self::Err> {
         const INVALID_PORT_NUMBER_MSG: &str = "invalid port number";
 
         match range.split_once('-') {
             None => {
                 let port = range
                     .parse::<NonZeroU16>()
-                    .map_err(|_| INVALID_PORT_NUMBER_MSG.to_string())?
+                    .map_err(|e| {
+                        Error::invalid_value(
+                            range,
+                            format!("{INVALID_PORT_NUMBER_MSG}: {e}"),
```

Comment:
similarly,  I think this would maybe be more "proper" use of `invalid_value` as 
```suggestion
                        Error::invalid_value(
                            "port_number",
                            format!("{range}: {e}"),
                        )
```
or something?

In particular, it's worth noting that the `fmt::Display` implementation for the `InvalidValue` variant is:

```rust
"Invalid Value: {label}, {message}"
```
so we probably don't need to repeat the words "invalid $THING" in the error message, since it's already there.

Reply:
The use of `Error::invalid_value` *should* now be following that pattern. I'd originally misinterpreted it as "value"->"err_msg", my bad.